sky/sdk/lib/editing/input.dart

```diff
@@ -64,10 +56,12 @@ class Input extends StatefulComponent {
 
   Widget build() {
     ThemeData themeData = Theme.of(this);
+    bool focused = Focus.at(this);
 
-    if (focused && !_isAttachedToKeyboard) {
-      keyboard.show(_editableValue.stub);
-      _isAttachedToKeyboard = true;
+    if (focused && !_keyboardHandle.attached) {
```

Comment:
build() is the hook Components have into the _sync() mechanism. _sync() is what happens when state changes. Focus is state. Seems to make sense to me. :-)

When would you do it?